parsl/dataflow/job_error_handler.py

```diff
@@ -0,0 +1,49 @@
+from typing import List, Any, Dict
+
+from parsl.dataflow.task_status_poller import ExecutorStatus
+from parsl.executors.base import ParslExecutor
+from parsl.providers.provider_base import JobStatus, JobState
+
+
+class JobErrorHandler(object):
+    def __init__(self):
+        pass
+
+    def run(self, status: List[ExecutorStatus]):
+        for es in status:
+            self._check_irrecoverable_executor(es)
+
+    def _check_irrecoverable_executor(self, es: ExecutorStatus):
+        if not es.executor.error_management_enabled:
+            return
+        custom_handling = es.executor.handle_errors(self, es.status)
+        if not custom_handling:
+            self.simple_error_handler(es.executor, es.status, 3)
+
+    def simple_error_handler(self, executor: ParslExecutor, status: Dict[Any, JobStatus], threshold: int):
+        (total_jobs, failed_jobs) = self.count_jobs(status)
+        if total_jobs >= threshold and failed_jobs == total_jobs:
```

Comment:
I worry that this test might fail if there are pending jobs in the queue. Since we can differentiate failed from completed with no errors, we could fail based on just `failed_jobs >= threshold`.

Reply:
This will not match if there are pending jobs in the queue. Do we want it to?

failed_jobs >= threshold would match eventually when you have mostly successful jobs and rare occasional failures. I don't think we want that. So maybe failed_job >= threshold and completed_jobs == 0 and running_jobs == 0.

Reply:
Since we don't know when pending jobs in the queue will be filled, we shouldn't have to wait when we know that `failed_jobs >= threshold`. We shouldn't need `complete_jobs == 0` and limit to cases where no progress is made, when it could be that say the allocation / quotes have been exceeded.

Reply:
See above: "failed_jobs >= threshold would match eventually when you have mostly successful jobs and rare occasional failures."

In other words, it is possible for failures to accumulate over time, even if the executor is mostly successful at running things.